pg_search/src/api/index.rs

```diff
@@ -690,6 +690,127 @@ range_term_fn!(
     true
 );
 
+#[derive(PostgresEnum, Serialize)]
+pub enum RangeRelation {
+    Intersects,
+    Contains,
+    Within,
+}
+
+impl Display for RangeRelation {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), std::fmt::Error> {
+        match self {
+            RangeRelation::Intersects => write!(f, "Intersects"),
+            RangeRelation::Contains => write!(f, "Contains"),
+            RangeRelation::Within => write!(f, "Within"),
+        }
+    }
+}
+
+macro_rules! range_term_range_fn {
+    ($func_name:ident, $value_type:ty, $is_datetime:expr, $default:expr) => {
+        #[pg_extern(name = "range_term", immutable, parallel_safe)]
+        pub fn $func_name(
+            field: FieldName,
+            range: $value_type,
+            relation: RangeRelation,
+        ) -> SearchQueryInput {
+            let (lower_bound, upper_bound) = match range.into_inner() {
+                None => (Bound::Included($default), Bound::Excluded($default)),
+                Some((lower, upper)) => {
+                    let lower_bound = match lower {
+                        RangeBound::Infinite => Bound::Unbounded,
+                        RangeBound::Inclusive(n) => Bound::Included(
+                            TantivyValue::try_from(n)
+                                .expect("value should be a valid TantivyValue representation")
+                                .tantivy_schema_value(),
+                        ),
+                        RangeBound::Exclusive(n) => Bound::Excluded(
+                            TantivyValue::try_from(n)
```

Comment:
This is a pretty big macro in terms of LOC. Generally you want to try and delegate out to functions and keep macros as small possible, ideally only containing templating logic.

If you found the type system for one reason or another was forcing you to make this one bigger, I understand. But I want to bring it up because large macros are harder to reason about and debug. They also make code navigation with LSP a little funky, and slow down compile times.

I think a macro was the right choice here, but considering that large parts of the body don't seem to reference any `$` templating, I thought I'd bring it up.